dspace-api/src/main/java/org/dspace/discovery/indexobject/IndexFactoryImpl.java

```diff
@@ -73,7 +73,11 @@ public SolrInputDocument buildDocument(Context context, T indexableObject) throw
     @Override
     public void writeDocument(Context context, T indexableObject, SolrInputDocument solrInputDocument)
             throws SQLException, IOException, SolrServerException {
-        writeDocument(solrInputDocument, null);
+        try {
+            writeDocument(solrInputDocument, null);
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
```

Comment:
Well, it might also be nice to know *what* error occurred.  We need the exception message *and* details that this class can provide.

But do we need the stack trace?

Reply:
I don't object to the stack trace; I wanted to make certain that we've considered whether it is useful.  If you know of a reason to include it, I am satisfied.  I've seen more than one case of logging hundreds of lines of backtrace that contain no useful information and I've grown cautious.

On further consideration, in this case one use would be to help us examine the execution path for the ultimate problem, which is that somewhere up the call stack there is code which discards exceptions without either reporting them or recovering from them.